presto-tests/src/test/java/com/facebook/presto/tests/TestEnumsWithNativeRewrite.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tpch.TpchPlugin;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.SystemSessionProperties.NATIVE_EXECUTION_ENABLED;
+import static com.facebook.presto.SystemSessionProperties.NATIVE_EXECUTION_TYPE_REWRITE_ENABLED;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static java.util.Collections.singletonList;
+
+@Test(singleThreaded = true)
+public class TestEnumsWithNativeRewrite
+        extends TestEnums
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        try {
+            Session session = testSessionBuilder()
+                    .setSystemProperty(NATIVE_EXECUTION_ENABLED, "true")
+                    .setSystemProperty(NATIVE_EXECUTION_TYPE_REWRITE_ENABLED, "true")
+                    .build();
+            DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session).setNodeCount(1).build();
+            queryRunner.enableTestFunctionNamespaces(ImmutableList.of("test"), ImmutableMap.of());
+            queryRunner.getMetadata().getFunctionAndTypeManager().addUserDefinedType(MOOD_ENUM);
+            queryRunner.getMetadata().getFunctionAndTypeManager().addUserDefinedType(COUNTRY_ENUM);
+            queryRunner.getMetadata().getFunctionAndTypeManager().addUserDefinedType(TEST_ENUM);
+            queryRunner.getMetadata().getFunctionAndTypeManager().addUserDefinedType(TEST_BIGINT_ENUM);
+            queryRunner.getMetadata().getFunctionAndTypeManager().addUserDefinedType(MARKET_SEGMENT_ENUM);
+
+            queryRunner.installPlugin(new TpchPlugin());
+            queryRunner.createCatalog("tpch", "tpch");
+            return queryRunner;
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Test
+    public void testEnumLiterals()
```

Comment:
instead of overriding all the tests, can you break them up in TestEnums so that the unsupported stuff are in separate tests. Then we can just override the unsupported ones, but ensure that we have the same coverage otherwise..

Reply:
Done!